models/Element/Service.php

```diff
@@ -1070,40 +1070,30 @@ public static function getValidKey(string $key, string $type): string
             'type' => $type,
         ]);
         \Pimcore::getEventDispatcher()->dispatch($event, SystemEvents::SERVICE_PRE_GET_VALID_KEY);
-        $key = $event->getArgument('key');
-        $key = trim($key);
-
-        // replace all control/unassigned and invalid characters
-        $key = preg_replace('/[^\PCc^\PCn^\PCs]/u', '', $key);
-        // replace all 4 byte unicode characters
-        $key = preg_replace('/[\x{10000}-\x{10FFFF}]/u', '-', $key);
-        // replace left to right marker characters ( lrm )
-        $key = preg_replace('/(\x{200e}|\x{200f})/u', '-', $key);
```

Comment:
I included this one in the `\p{C}` one, because they're "format" characters, which are in the `\p{Cf}` group.